library/lib_num_to_formatted_str.ks

```diff
@@ -40,10 +40,14 @@ LOCAL FUNCTION time_string {
   PARAMETER timeSec,    // the time in seconds to format
   fixedPlaces,          // number of required time places (e.g. HH:MM:SS has 3 fixedPlaces)
   stringList,           // separators for each time place (must have at least fixedPlaces elements!)
-  roundingList,         // rounding factors for each time place (should match length of stringList)
+  rounding,             // rounding for the seconds place, range 0 to 2
   prependT,             // prepend a T- or T+ to format (T- or T if showPlus is FALSE)
   showPlus IS prependT. // by default only display "+" when prependT is TRUE
 
+  // start by rounding the input so we don't have to "carry the one" in time_converter
+  LOCAL roundingList IS LIST(MIN(rounding,2), 0, 0, 0, 0).
+  SET timeSec TO ROUND(timeSec, rounding).
```

Comment:
```suggestion
  SET timeSec TO ROUND(timeSec, roundingList[0]).
```
This rounding operation also needs to respect the no larger than 2 limit or else if the user passed in a larger number we could still see the error you where trying to fix.